packages/itwin/ec3-widget/src/components/CreateAssemblyComponent.tsx

```diff
@@ -0,0 +1,333 @@
+    const existingAssembly = reportTableLabels.find((x) => x.value === assembly.reportTable);
+    if (existingAssembly) {
+      const allAssem = reportTableLabels.filter((x) => !allAssemblies?.map((p) => p.reportTable).includes(x.value));
+      allAssem.push(existingAssembly);
+      return allAssem;
+    }
+    return reportTableLabels.filter((x) => !allAssemblies?.map((p) => p.reportTable).includes(x.value));
+  };
+
+  return (
+    <div className="ec3w-create-assembly-step">
+      {isLoading ? (
+        <ProgressRadial className="ec3w-loading-indicator" indeterminate />
+      ) : (
+        <>
+          <div className="ec3w-assembly-list">
+            {allAssemblies &&
+              allAssemblies.length > 0 &&
+              allAssemblies?.map((assembly, i) => {
+                return (
+                  <>
+                    <ExpandableBlock
+                      className="ec3w-assembly-expandable-block"
+                      title={`${assembly.name}`}
+                      endIcon={
+                        <ButtonGroup>
+                          <IconButton
+                            key={`edit-assembly${i}`}
+                            styleType="borderless"
+                            onClick={(e) => {
+                              setEditableAssemblyIndex(i);
+                              e.stopPropagation();
+                            }}
+                          >
+                            <SvgEdit />
+                          </IconButton>
+                          <IconButton
+                            key={`delete-assembly${i}`}
+                            disabled={allAssemblies.length === 1}
+                            styleType="borderless"
+                            onClick={(e) => {
+                              e.stopPropagation();
+                              onAssemblyDataChange(assembly, i, "delete");
+                            }}
+                          >
+                            <SvgDelete />
+                          </IconButton>
+                        </ButtonGroup>
+                      }
+                      key={i}
+                      isExpanded={editableAssemblyIndex ? i === editableAssemblyIndex : i === 0}
+                    >
+                      <>
+                        <LabeledInput
+                          id="name"
+                          name="name"
+                          label="Assembly Name"
+                          value={assembly.name}
+                          onChange={(event) => {
+                            onAssemblyDataChange({ ...assembly, name: event.target.value }, i);
+                          }}
+                          disabled={i !== editableAssemblyIndex}
+                        />
+                        <LabeledSelect
+                          label="Select ReportTable"
+                          data-testid="ec3-report-table-select"
+                          options={getReportTableOptions(assembly)}
```

Comment:
This looks like it will run at every rerender.

Reply:
done